hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java

```diff
@@ -361,42 +360,32 @@ public void commit() {
 			final HederaWorldState wrapped = (HederaWorldState) wrappedWorldView();
 			final var entityAccess = wrapped.entityAccess;
 
+			commitSizeLimitedStorageTo(entityAccess);
+
 			/* Note that both the adjustBalance() and spawn() calls in the blocks below are ONLY
 			 * needed to make sure the record's ℏ transfer list is constructed properly---the
 			 * finishing call to trackingLedgers().commits() at the end of this method will persist
 			 * all the same information. */
 			final var deletedAddresses = getDeletedAccountAddresses();
 			deletedAddresses.forEach(address -> {
 				final var accountId = accountParsedFromSolidityAddress(address);
+				ensureExistence(accountId, entityAccess, wrapped.provisionalContractCreations);
 				final var deletedBalance= entityAccess.getBalance(accountId);
 				entityAccess.adjustBalance(accountId, -deletedBalance);
 			});
 			for (final var updatedAccount : getUpdatedAccounts()) {
 				final var accountId = accountParsedFromSolidityAddress(updatedAccount.getAddress());
 
-				if (!entityAccess.isExtant(accountId)) {
-					wrapped.provisionalContractCreations.add(asContract(accountId));
-					entityAccess.spawn(accountId, 0L, CONTRACT_CUSTOMIZER);
-				}
+				ensureExistence(accountId, entityAccess, wrapped.provisionalContractCreations);
```

Comment:
should we update the kvUpdates in here instead of at the beginning of the commit ?